### PR TITLE
Rust: More test cases for unused variables

### PR DESCRIPTION
**Pull request by geoffw0**
More test cases for unused variables.  Confirms that the issue producing spurious results for `None` matches is due to use of unqualified enum values that are being misrecognized as variables.

### CODE REVIEW

File: rust/ql/test/query-tests/unusedentities/main.rs [hunk 1]
```diff
@@ -228,18 +235,87 @@ fn if_lets() {
...
+
+    let h = YesOrNo::Yes;
+    match h {
         YesOrNo::Yes => {}
         YesOrNo::No => {}
     }
+
+    let i = Yes;
+    match i {
+        Yes => {} // SPURIOUS: unused variable 'None'
```
Comment (hvitved) [inline rust/ql/test/query-tests/unusedentities/main.rs:268]: 'None' -> 'Yes'

File: rust/ql/test/query-tests/unusedentities/main.rs [hunk 2]
```diff
@@ -228,18 +235,87 @@ fn if_lets() {
...
+        No => {} // SPURIOUS: unused variable 'None'
```
Comment (hvitved) [inline rust/ql/test/query-tests/unusedentities/main.rs:269]: 'None' -> 'No'

**Review by hvitved — CHANGES_REQUESTED** [review]
Two minor things, otherwise LGTM

File: rust/ql/test/query-tests/unusedentities/main.rs (hunk 1 above)
Comment (geoffw0) [reply rust/ql/test/query-tests/unusedentities/main.rs]: Oops, well spotted.  Now fixed.

**Comment by geoffw0** [comment]
Corrected minor mistakes and added one more test case.

**Review by hvitved — APPROVED** [review]

**Pull request merged**